datascroller/scroller.py

```diff
@@ -447,74 +452,75 @@ def key_press_and_print_df(stdscr, df):
-        elif key in [SCROLL_DOWN, curses.KEY_DOWN]:
+        elif key in [keys.SCROLL_DOWN, curses.KEY_DOWN]:
             df_window.move_down()
-        elif key in [SCROLL_UP, curses.KEY_UP]:
+        elif key in [keys.SCROLL_UP, curses.KEY_UP]:
             df_window.move_up()
 
         # Moving fast
-        elif key == PAGE_DOWN:
+        elif key == keys.PAGE_DOWN:
             df_window.page_down()
-        elif key == PAGE_UP:
+        elif key == keys.PAGE_UP:
             df_window.page_up()
 
         # alternate views
-        elif key == HIGHLIGHT:
+        elif key == keys.HIGHLIGHT:
             df_window.toggle_highlight_mode()
 
-        elif key == HELP:
-            # help_view = not help_view
-            pass
+        elif key == keys.HELP:
+            help_view = not help_view
 
         # search functionality
-        elif key == LINE_SEARCH:
+        elif key == keys.LINE_SEARCH:
             search_string = get_user_input_with_prompt(stdscr, term_rows - 1, 0,
                                                        "Goto line: ")
             if len(search_string) > 0:
                 try:
                     df_window.line_search(int(search_string))
                 except ValueError:
-                    pass
-                    # TODO(johncmerfeld): Reprimand the user?
+                    err_string = "Please enter a valid (integer) line number"
 
-        elif key == FILTER:
+        elif key == keys.FILTER:
             query_bytes = get_user_input_with_prompt(stdscr, term_rows - 1, 0,
                                                      "Column filter: ")
             query_string = query_bytes.decode(encoding="utf-8")
             if len(query_string) > 0:
-                try:
-                    df_window = df_window.filter(query_string, viewing_area)
-                except sqldf.PandaSQLException:  # TODO better exception handling
-                    pass
-                    # TODO(johncmerfeld): Reprimand the user?
+                df_window = df_window.filter(query_string, viewing_area)
+                # NOTE: if column names are invalid,
+                # an empty dataframe is returned.
+                # We could preempt this if we wanted.
 
-        elif key == QUERY:
+        elif key == keys.QUERY:
             query_bytes = get_user_input_with_prompt(stdscr, term_rows - 1, 0,
                                                      "SQL query (use 'df' as table name): ")
             query_string = query_bytes.decode(encoding="utf-8")
             if len(query_string) > 0:
                 try:
                     df_window = df_window.query(query_string, viewing_area)
-                except SyntaxError:
-                    pass
-                    # TODO(johncmerfeld): Reprimand the user?
+                except:  # noqa: E722
+                    err_string = "Syntax error in query"
+                    # TODO(johncmerfeld): be more specific with this error
```

Comment:
There's got to be some way to tap into the message coming from sqldf. But for a first pass, a message to the user stating that there was something wrong with the query and a chance to try again is probably more than enough.

Reply:
I think it can be done but it just was not working for me. Additionally, the message that pandasql does generate is unhelpful. It's the same for every SQL error and doesn't tell you where the mistake is, whether it's syntax or a missing table/column, etc. So this might be an area where we're better off writing our own stuff.